### PR TITLE
CM-417: Update tekton pipeline config code for istio-csr

### PR DESCRIPTION
**Pull request by anirudhAgniRedhat**
This PR updates the tekton pipeline config code for cert-manager-istio-csr files.

cc: @bharath-b-rh

### CODE REVIEW

File: .tekton/cert-manager-istio-csr-1-15-push.yaml [hunk 1]
```diff
@@ -108,7 +119,7 @@ spec:
       name: build-args-file
       type: string
     - default:
-      - linux/x86_64
+      - $(params.build-platforms)
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-push.yaml:122]: This is missed in pull-request yaml, please update it.

Same is missed in `https://github.com/openshift/cert-manager-operator-release/blob/release-1.15/.tekton/cert-manager-operator-1-15-pull-request.yaml` . If you don't mind, could you please update it as well, but with a different commit.

File: .tekton/cert-manager-istio-csr-1-15-push.yaml [hunk 2]
```diff
@@ -3,11 +3,14 @@ kind: PipelineRun
...
+    build.appstudio.openshift.io/pipeline: '{"name":"docker-build-multi-platform-oci-ta","bundle":"latest"}'
+    build.appstudio.openshift.io/request: "configure-pac"
     build.appstudio.redhat.com/commit_sha: '{{revision}}'
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
-    pipelinesascode.tekton.dev/on-cel-expression: event == "push" && target_branch
-      == "release-1.15"
+    pipelinesascode.tekton.dev/on-cel-expression: event == "push" && target_branch == "release-1.15" &&
+      (".tekton/cert-manager-operator-istio-csr-1-15-push.yaml".pathChanged() || "Containerfile.cert-manager-operator-istio-csr".pathChanged() ||
+      "cert-manager-operator-istio-csr/***".pathChanged())
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-push.yaml:13]:
```suggestion
      "cert-manager-istio-csr/***".pathChanged())
```

File: .tekton/cert-manager-istio-csr-1-15-push.yaml [hunk 3]
```diff
@@ -3,11 +3,14 @@ kind: PipelineRun
...
     build.appstudio.openshift.io/repo: https://github.com/openshift/cert-manager-operator-release?rev={{revision}}
+    build.appstudio.openshift.io/pipeline: '{"name":"docker-build-multi-platform-oci-ta","bundle":"latest"}'
+    build.appstudio.openshift.io/request: "configure-pac"
     build.appstudio.redhat.com/commit_sha: '{{revision}}'
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
-    pipelinesascode.tekton.dev/on-cel-expression: event == "push" && target_branch
-      == "release-1.15"
+    pipelinesascode.tekton.dev/on-cel-expression: event == "push" && target_branch == "release-1.15" &&
+      (".tekton/cert-manager-operator-istio-csr-1-15-push.yaml".pathChanged() || "Containerfile.cert-manager-operator-istio-csr".pathChanged() ||
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-push.yaml:12]:
```suggestion
      (".tekton/cert-manager-istio-csr-1-15-push.yaml".pathChanged() || "Containerfile.cert-manager-istio-csr".pathChanged() ||
```

File: .tekton/cert-manager-istio-csr-1-15-pull-request.yaml [hunk 4]
```diff
@@ -3,12 +3,15 @@ kind: PipelineRun
...
+    build.appstudio.openshift.io/request: "configure-pac"
     build.appstudio.redhat.com/commit_sha: '{{revision}}'
     build.appstudio.redhat.com/pull_request_number: '{{pull_request_number}}'
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
-    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
-      == "release-1.15"
+    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch == "release-1.15" &&
+      (".tekton/cert-manager-operator-istio-csr-1-15-pull-request.yaml".pathChanged() || "Containerfile.cert-manager-operator-istio-csr".pathChanged() ||
+      "cert-manager-operator-istio-csr/***".pathChanged())
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-pull-request.yaml:14]:
```suggestion
      "cert-manager-istio-csr/***".pathChanged())
```

File: .tekton/cert-manager-istio-csr-1-15-pull-request.yaml [hunk 5]
```diff
@@ -3,12 +3,15 @@ kind: PipelineRun
...
+    build.appstudio.openshift.io/pipeline: '{"name":"docker-build-multi-platform-oci-ta","bundle":"latest"}'
+    build.appstudio.openshift.io/request: "configure-pac"
     build.appstudio.redhat.com/commit_sha: '{{revision}}'
     build.appstudio.redhat.com/pull_request_number: '{{pull_request_number}}'
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
-    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
-      == "release-1.15"
+    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch == "release-1.15" &&
+      (".tekton/cert-manager-operator-istio-csr-1-15-pull-request.yaml".pathChanged() || "Containerfile.cert-manager-operator-istio-csr".pathChanged() ||
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-pull-request.yaml:13]:
```suggestion
      (".tekton/cert-manager-istio-csr-1-15-pull-request.yaml".pathChanged() || "Containerfile.cert-manager-istio-csr".pathChanged() ||
```

File: .tekton/cert-manager-istio-csr-1-15-push.yaml (hunk 1 above)
Comment (anirudhAgniRedhat) [reply .tekton/cert-manager-istio-csr-1-15-push.yaml:122]: Thanks @bharath-b-rh For pointing this out Added the suggestion in a different commit!

File: .tekton/cert-manager-istio-csr-1-15-push.yaml [hunk 6]
```diff
@@ -23,13 +26,23 @@ spec:
...
+      - linux/s390x
+      - linux/ppc64le
+      - linux/arm64
+  - name: build-args
+    value:
+      - "RELEASE_VERSION=v1.15.0"
+      - "COMMIT_SHA={{revision}}"
+      - "SOURCE_URL={{source_url}}"
+  - name: prefetch-input
+    value: '{"type": "gomod", "path": "."}'
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-push.yaml:45]:
```suggestion
    value: '{"type": "gomod", "path": "cert-manager-istio-csr"}'
```

File: .tekton/cert-manager-istio-csr-1-15-pull-request.yaml [hunk 7]
```diff
@@ -26,13 +29,23 @@ spec:
...
+      - linux/s390x
+      - linux/ppc64le
+      - linux/arm64
+  - name: build-args
+    value:
+      - "RELEASE_VERSION=v1.15.0"
+      - "COMMIT_SHA={{revision}}"
+      - "SOURCE_URL={{source_url}}"
+  - name: prefetch-input
+    value: '{"type": "gomod", "path": "."}'
```
Comment (bharath-b-rh) [inline .tekton/cert-manager-istio-csr-1-15-pull-request.yaml:48]:
```suggestion
    value: '{"type": "gomod", "path": "cert-manager-istio-csr"}'
```

**Comment by anirudhAgniRedhat** [comment]
```
Go compliance shim [216] [rhel-9-golang-1.22][openshift-golang-builder]: Exited with: 0
[2/2] STEP 1/10: FROM registry.redhat.io/rhel9-4-els/rhel:9.4
Timeout, server 10.241.72.194 not responding.
```

Error due to timeout!!

/retest

**Comment by bharath-b-rh** [comment]
istio-csr and cert-manager operator EC policy validations are failing because of vulnerabilities.

istio-csr validation
```
{
	"msg": "Found packages with critical vulnerabilities. Consider updating to a newer version of those packages, they may no longer be affected by the reported CVEs.",
	"metadata": {
		"details": {
			"description": "Vulnerabilities found: golang.org/x/crypto-v0.26.0 (GHSA-v778-237x-gjrc)",
			"name": "clair_critical_vulnerabilities",
			"url": "https://access.redhat.com/articles/red_hat_vulnerability_tutorial"
		},
		"vulnerabilities_number": 1
	}
},
{
	"msg": "Found packages with high vulnerabilities. Consider updating to a newer version of those packages, they may no longer be affected by the reported CVEs.",
	"metadata": {
		"details": {
			"description": "Vulnerabilities found: istio.io/istio-v0.0.0-20240821191124-89d73dc1f639 (GHSA-7774-7vr3-cc8j, GHSA-856q-xv3c-7f2f, GHSA-hqxw-mm44-gc4r, GHSA-qcvw-82hh-gq38), golang.org/x/net-v0.28.0 (GHSA-w32m-9786-jp63)",
			"name": "clair_high_vulnerabilities",
			"url": "https://access.redhat.com/articles/red_hat_vulnerability_tutorial"
		},
		"vulnerabilities_number": 5
	}
},
{
	"msg": "Found packages with medium vulnerabilities. Consider updating to a newer version of those packages, they may no longer be affected by the reported CVEs.",
	"metadata": {
		"details": {
			"description": "Vulnerabilities found: python3-libs-3.9.18-3.el9_4.6 (CVE-2024-9287), istio.io/istio-v0.0.0-20240821191124-89d73dc1f639 (GHSA-xwx5-5c9g-x68x), github.com/cert-manager/cert-manager-v1.15.3 (GHSA-r4pg-vg54-wxx4), python3-3.9.18-3.el9_4.6 (CVE-2024-9287)",
			"name": "clair_medium_vulnerabilities",
			"url": "https://access.redhat.com/articles/red_hat_vulnerability_tutorial"
		},
		"vulnerabilities_number": 4
	}
},
{
	"msg": "Found packages with unknown vulnerabilities. Consider updating to a newer version of those packages, they may no longer be affected by the reported CVEs.",
	"metadata": {
		"details": {
			"description": "Vulnerabilities found: golang.org/x/net-v0.28.0 (GO-2024-3333), golang.org/x/crypto-v0.26.0 (GO-2024-3321)",
			"name": "clair_unknown_vulnerabilities",
			"url": "https://access.redhat.com/articles/red_hat_vulnerability_tutorial"
		},
		"vulnerabilities_number": 2
	}
}
```

**Comment by bharath-b-rh** [comment]
overriding EC policies failures for now, to discuss and work on the vulernabilities.

/override "Red Hat Konflux / cert-manager-istio-csr-1-15-enterprise-contract / cert-manager-istio-csr-1-15"
/override "Red Hat Konflux / cert-manager-operator-1-15-on-pull-request"

**Pull request merged**